controllers/remote/restconfig.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remote
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	gruntime "runtime"
+	"strings"
+
+	"sigs.k8s.io/cluster-api/cmd/version"
+)
+
+const (
+	unknowString = "unknown"
+)
+
+func buildUserAgent(command, version, os, arch, commit string) string {
+	return fmt.Sprintf(
+		"%s/%s (%s/%s) cluster.x-k8s.io/%s", command, version, os, arch, commit)
+}
+
+// DefaultClusterAPIUserAgent returns a User-Agent string built from static global vars.
+func DefaultClusterAPIUserAgent() string {
+	return buildUserAgent(
+		adjustCommand(os.Args[0]),
```

Comment:
Based on the current scaffolding that we are using, wouldn't this be `/manager` for all deployments?

https://github.com/kubernetes-sigs/cluster-api/blob/953ad253ee2bc169ddda765efe5851c30ffb1def/config/manager/manager.yaml#L21
https://github.com/kubernetes-sigs/cluster-api/blob/953ad253ee2bc169ddda765efe5851c30ffb1def/bootstrap/kubeadm/config/manager/manager.yaml#L20
https://github.com/kubernetes-sigs/cluster-api/blob/953ad253ee2bc169ddda765efe5851c30ffb1def/controlplane/kubeadm/config/manager/manager.yaml#L20

Reply:
should we not have a command?

Reply:
I don't think having the command is necessarily a bad thing, just that it isn't sufficient to differentiate between either controllers using the remote client or even which specific controller manager is using the remote client.

Reply:
I'm not sure how to add the component as well, will look into it

Reply:
marking as resolved since source name is now being added